src/plugins/auto/plugin.cpp

```diff
@@ -640,7 +640,20 @@ QueryNetworkResult MultiDeviceInferencePlugin::QueryNetwork(const CNNNetwork&
     auto queryconfig = _pluginConfig;
     // updateFromMap will check config valid
     queryconfig.UpdateFromMap(config, GetName(), true);
+    bool workModeAuto = GetName() == "AUTO";
     auto fullConfig = queryconfig._keyConfigMap;
+    if (!queryconfig._isSetPerHint) {
+        fullConfig.erase(PluginConfigParams::KEY_PERFORMANCE_HINT);
+        if (workModeAuto) {
+            // set performance hint to 'LATENCY' model for AutoExecutable Network.
+            queryconfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT,
+                                                   PluginConfigParams::LATENCY);
+        } else {
+            // set performance hint to 'THROUGHPUT' model for MultiExecutable Network.
+            queryconfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT,
+                                                   PluginConfigParams::THROUGHPUT);
+        }
+    }
```

Comment:
why we have to update this perf hint setting in queryNetwork

Reply:
Removed now.

Reply:
This logic is redundant after initializing performance mode config of AUTO plugin with default value "UNDEFINED" .